components/molecules/HighlightInput/highlight-input-form.tsx

```diff
@@ -432,6 +432,12 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
 
       const taggedRepoFullNames = taggedRepoList.map((repo) => `${repo.repoOwner}/${repo.repoName}`);
 
+      // Check if the user has tagged at least one repo and ask them to tag at least one if they haven't
+      if (taggedRepoList.length < 1) {
+        setError("Please tag at least one repo!");
```

Comment:
wording (blocking): Make it clear why at least one repository needs to be added.

```suggestion
        setError("Please add at least one repository associated with the blog post");
```

@BekahHW, any suggestions for better wording?

Reply:
Ok, I think this is tricky. I need some clarification first. The problem is that when someone submits a blog post, they are automatically generating a repo, right?

I think the repo should be optional in for blog post highlights. For example, it's still valid to highlight work on something like "A guide to writing documentation." There's not necessarily a repo associated with that. 

Can we conditionally check if the highlight is a blog post, and then leave the input empty unless they put something in there?

Reply:
@bdougie thoughts on the above?

Reply:
If we move forward with this implementation, I think Nick's copy works just change the to your:

> "Please add at least one repository associated with your blog post"

Reply:
Good point that it needs to be when a blog post is added @BekahHW. Missed that. 😅 

@shamimbinnur, we'd need to check that it's a blog post before showing that error.

Reply:
Hey @nickytonline, thanks for suggesting the better wording.

Not allowing the user to post a blog without tagging the repo is intentional. As @bdougie's comment [here](https://github.com/open-sauced/app/issues/1995#issue-1959992628) states, "every blog highlight should be linked to a repo."

If no repo is tagged whether automatically or manually an empty array of repos is passed to the backend. However, the backend does not keep it empty; it just splits text from the blog link and creates an invalid repo URL.

Let me be clear, "Does every blog post highlight need to be linked to a repo?"

CC: @BekahHW @bdougie

Reply:
Ok, so you're saying from a backend perspective, we _have_ to have a repo listed?